File: Ruby/Civitas/lib/casilla_juez.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module Civitas
  class CasillaJuez < Casilla
    def initialize(name, num_casilla_carcel)
      super(name)
      @carcel=num_casilla_carcel
    end
  
    # Método sobrescrito
    def recibe_jugador(actual, todos)
      if(jugador_correcto(actual, todos))
        informe(actual, todos)
        todos[actual].encarcelar(@carcel)
      end
    end
  
    
    def to_s
      "\n\t DESCRIPCIÓN CASILLA: \n
        \n Nombre: #{nombre}
        \n Tipo: JUEZ
        \n Carcel en la casilla: #{@carcel}
        "
    end
  end
end

File: Java/Civitas/src/civitas/CasillaJuez.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author arturo
 */
public class CasillaJuez extends Casilla{
    private static int numCasillaCarcel;
    
    CasillaJuez(int numCasillaCarcel, String nombre){
        super(nombre);
        this.numCasillaCarcel=numCasillaCarcel;
    }
    
    @Override
    public void recibeJugador(int actual, ArrayList<Jugador> todos){ 
        if(jugadorCorrecto(actual, todos)==true){
            this.informe(actual, todos);
            todos.get(actual).encarcelar(numCasillaCarcel); 
        }
    }
    
}